src/xtgeo/grid3d/_grid_import.py

```diff
@@ -80,13 +42,16 @@ def from_file(gfile, fformat=None, **kwargs):  # pylint: disable=too-many-branch
         result.update(_grid_import_ecl.import_ecl_grdecl(gfile, **kwargs))
     elif fformat == "bgrdecl":
         result.update(_grid_import_ecl.import_ecl_bgrdecl(gfile, **kwargs))
-    elif fformat == "xtgf":
+    elif fformat == "xtg":
         result.update(_grid_import_xtgcpgeom.import_xtgcpgeom(gfile, **kwargs))
     elif fformat == "hdf":
         result.update(_grid_import_xtgcpgeom.import_hdf5_cpgeom(gfile, **kwargs))
     else:
-        raise ValueError("Invalid file format")
+        raise ValueError(f"Invalid file format: {fformat}")
 
-    result["name"] = gfile.file.stem
+    if gfile.memstream:
+        result["name"] = "unknown"
```

Comment:
None is used to indicate not from file elsewhere, I think that would be better

Reply:
The string `unknown` is applied as this is consistent with e.g. `RegularSurface`'s `name` attribute when `name` is not known.